ide/jdt/test/src/test/java/com/ibm/wala/demandpa/driver/DemandCastChecker.java

```diff
@@ -1,301 +0,0 @@
- * SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
- */
-package com.ibm.wala.demandpa.driver;
-
-import com.ibm.wala.classLoader.Language;
-import com.ibm.wala.core.tests.callGraph.CallGraphTestUtil;
-import com.ibm.wala.core.tests.util.TestConstants;
-import com.ibm.wala.core.util.ProgressMaster;
-import com.ibm.wala.demandpa.alg.ContextSensitiveStateMachine;
-import com.ibm.wala.demandpa.alg.DemandRefinementPointsTo;
-import com.ibm.wala.demandpa.alg.DemandRefinementPointsTo.PointsToResult;
-import com.ibm.wala.demandpa.alg.refinepolicy.FieldRefinePolicy;
-import com.ibm.wala.demandpa.alg.refinepolicy.ManualFieldPolicy;
-import com.ibm.wala.demandpa.alg.refinepolicy.ManualRefinementPolicy;
-import com.ibm.wala.demandpa.alg.refinepolicy.RefinementPolicyFactory;
-import com.ibm.wala.demandpa.alg.refinepolicy.TunedRefinementPolicy;
-import com.ibm.wala.demandpa.alg.statemachine.StateMachineFactory;
-import com.ibm.wala.demandpa.flowgraph.IFlowLabel;
-import com.ibm.wala.demandpa.util.MemoryAccessMap;
-import com.ibm.wala.demandpa.util.SimpleMemoryAccessMap;
-import com.ibm.wala.ipa.callgraph.AnalysisCacheImpl;
-import com.ibm.wala.ipa.callgraph.AnalysisOptions;
-import com.ibm.wala.ipa.callgraph.AnalysisScope;
-import com.ibm.wala.ipa.callgraph.CGNode;
-import com.ibm.wala.ipa.callgraph.CallGraph;
-import com.ibm.wala.ipa.callgraph.CallGraphBuilder;
-import com.ibm.wala.ipa.callgraph.CallGraphBuilderCancelException;
-import com.ibm.wala.ipa.callgraph.CallGraphStats;
-import com.ibm.wala.ipa.callgraph.Entrypoint;
-import com.ibm.wala.ipa.callgraph.IAnalysisCacheView;
-import com.ibm.wala.ipa.callgraph.impl.Util;
-import com.ibm.wala.ipa.callgraph.propagation.HeapModel;
-import com.ibm.wala.ipa.callgraph.propagation.InstanceKey;
-import com.ibm.wala.ipa.callgraph.propagation.PointerAnalysis;
-import com.ibm.wala.ipa.callgraph.propagation.PointerKey;
-import com.ibm.wala.ipa.cha.ClassHierarchy;
-import com.ibm.wala.ipa.cha.ClassHierarchyException;
-import com.ibm.wala.ipa.cha.ClassHierarchyFactory;
-import com.ibm.wala.ipa.cha.IClassHierarchy;
-import com.ibm.wala.properties.WalaProperties;
-import com.ibm.wala.ssa.IR;
-import com.ibm.wala.ssa.SSACheckCastInstruction;
-import com.ibm.wala.ssa.SSAInstruction;
-import com.ibm.wala.types.ClassLoaderReference;
-import com.ibm.wala.types.TypeReference;
-import com.ibm.wala.util.CancelException;
-import com.ibm.wala.util.MonitorUtil.IProgressMonitor;
-import com.ibm.wala.util.NullProgressMonitor;
-import com.ibm.wala.util.WalaException;
-import com.ibm.wala.util.collections.Pair;
-import com.ibm.wala.util.debug.Assertions;
-import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.List;
-import java.util.Properties;
-import java.util.function.Predicate;
-
-/**
- * Uses a demand-driven points-to analysis to check the safety of downcasts.
- *
- * @author Manu Sridharan
- */
-public class DemandCastChecker {
```

Comment:
This code I would like to keep around as it's a useful example.  Could it be moved to `core/src/main/java/com/ibm/wala/examples/drivers`?

Reply:
Retained and moved in the latest version (c359fb1ef6a541e2373967607cf6abf064da86e7) of this pull request.